client/src/components/organisms/beacons/create-beacon-page.tsx

```diff
@@ -77,9 +85,20 @@ const CreateBeaconPage: FC<{ user: User }> = ({ user }) => {
       await createBeacon(draftData).unwrap();
       setDraftSaved(true);
       setTimeout(() => setDraftSaved(false), 3000);
-    } catch (error) {
+    } catch (error: any) {
       console.error("Failed to save draft:", error);
-      setDraftError("Failed to save draft. Please try again.");
+      
+      if (error?.data?.errors) {
+        if (error.data.errors.CategoryId) {
+          setDraftError("Server error: Category validation failed. Try again or select a category.");
+        } else if (error.data.errors.ItemName) {
+          setDraftError("Please provide a title for your beacon draft.");
+        } else {
+          setDraftError("There was an issue with your beacon draft. Please check all fields and try again.");
+        }
+      } else {
+        setDraftError("Failed to save draft. Please check your connection and try again.");
+      }
```

Comment:
move this into a helper function in a different file and simplify:


```suggestion
        if (!error?.data?.errors) {
          setDraftError("Failed to save draft. Please check your connection and try again.");
        }
        if (error.data.errors.CategoryId) {
          setDraftError("Server error: Category validation failed. Try again or select a category.");
        } else if (error.data.errors.ItemName) {
          setDraftError("Please provide a title for your beacon draft.");
        } else {
          setDraftError("There was an issue with your beacon draft. Please check all fields and try again.");
        }
```